be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -1,44 +1,89 @@
 package ch.epfl.pop.pubsub.graph
 
 import java.util.concurrent.TimeUnit
+import java.io.File
 
-import akka.actor.typed.ActorRef
+import akka.actor.{Actor, ActorLogging}
+import akka.event.LoggingReceive
 import akka.pattern.AskableActorRef
 import akka.util.Timeout
 import ch.epfl.pop.model.network.method.message.Message
 import ch.epfl.pop.model.objects.{Channel, Hash}
+import ch.epfl.pop.pubsub.PublishSubscribe
+import org.iq80.leveldb.impl.Iq80DBFactory.factory
+import org.iq80.leveldb.{DB, Options}
 
+import scala.collection.mutable
 import scala.concurrent.duration.{Duration, FiniteDuration}
 
-// FIXME will remove DBActor once everything is working
 object DbActor {
 
+  final val DATABASE_FOLDER: String = "database"
   final val DURATION: FiniteDuration = Duration(1, TimeUnit.SECONDS)
   final val TIMEOUT: Timeout = Timeout(DURATION)
-  final lazy val INSTANCE: AskableActorRef = ??? // FIXME created at server launch
+  final lazy val INSTANCE: AskableActorRef = PublishSubscribe.getDbActorRef
 
-  sealed trait DbMessage
+  sealed trait Event
 
   /**
    * Request to write a message in the database
    * @param channel the channel where the message must be published
    * @param message the message to write in the database
-   * @param replyTo the actor to respond to
    */
-  final case class Write(channel: Channel, message: Message, replyTo: ActorRef[Boolean]) extends DbMessage
+  final case class Write(channel: Channel, message: Message) extends Event
 
   /**
    * Request to read a specific messages on a channel
    * @param channel the channel where the message was published
    * @param id the id of the message we want to read
-   * @param replyTo the actor to reply to
    */
-  final case class Read(channel: Channel, id: Hash, replyTo: ActorRef[Option[Message]]) extends DbMessage
+  final case class Read(channel: Channel, id: Hash) extends Event
+
+
+  sealed trait DbActorMessage
+
+  final case object DbActorWriteAck extends DbActorMessage
+  final case object DbActorReadAck extends DbActorMessage
+  final case class DbActorNAck(code: Int, description: String) extends DbActorMessage
 
 
   def getInstance: AskableActorRef = INSTANCE
 
   def getTimeout: Timeout = TIMEOUT
 
   def getDuration: FiniteDuration = DURATION
+
+  def apply(): DbActor = DbActor()
+
+  sealed case class DbActor() extends Actor with ActorLogging {
```

Comment:
@nkcr created a relational database schema in be1 to store all the data.
Would it be a useful inspiration? I'm available to discuss